src/irr/asset/CGLILoader.cpp

```diff
@@ -0,0 +1,457 @@
+				case gl::INTERNAL_LA16: return getTranslatedFinalFormat();				//GL_LUMINANCE16_ALPHA16
+
+				// Depth formats
+				case gl::INTERNAL_D16: return getTranslatedFinalFormat();				//GL_DEPTH_COMPONENT16
+				case gl::INTERNAL_D24: return getTranslatedFinalFormat();				//GL_DEPTH_COMPONENT24
+				case gl::INTERNAL_D16S8_EXT: return getTranslatedFinalFormat();
+				case gl::INTERNAL_D24S8: return getTranslatedFinalFormat();			//GL_DEPTH24_STENCIL8
+				case gl::INTERNAL_D32: return getTranslatedFinalFormat();				//GL_DEPTH_COMPONENT32
+				case gl::INTERNAL_D32F: return getTranslatedFinalFormat();				//GL_DEPTH_COMPONENT32F
+				case gl::INTERNAL_D32FS8X24: return getTranslatedFinalFormat();		//GL_DEPTH32F_STENCIL8
+				case gl::INTERNAL_S8_EXT: return getTranslatedFinalFormat();			//GL_STENCIL_INDEX8
+
+				// Compressed formats
+				case gl::INTERNAL_RGB_DXT1: return getTranslatedFinalFormat();						//GL_COMPRESSED_RGB_S3TC_DXT1_EXT
+				case gl::INTERNAL_RGBA_DXT1: return getTranslatedFinalFormat();					//GL_COMPRESSED_RGBA_S3TC_DXT1_EXT
+				case gl::INTERNAL_RGBA_DXT3: return getTranslatedFinalFormat();					//GL_COMPRESSED_RGBA_S3TC_DXT3_EXT
+				case gl::INTERNAL_RGBA_DXT5: return getTranslatedFinalFormat();					//GL_COMPRESSED_RGBA_S3TC_DXT5_EXT
+				case gl::INTERNAL_R_ATI1N_UNORM: return getTranslatedFinalFormat();				//GL_COMPRESSED_RED_RGTC1
+				case gl::INTERNAL_R_ATI1N_SNORM: return getTranslatedFinalFormat();				//GL_COMPRESSED_SIGNED_RED_RGTC1
+				case gl::INTERNAL_RG_ATI2N_UNORM: return getTranslatedFinalFormat();				//GL_COMPRESSED_RG_RGTC2
+				case gl::INTERNAL_RG_ATI2N_SNORM: return getTranslatedFinalFormat();				//GL_COMPRESSED_SIGNED_RG_RGTC2
+				case gl::INTERNAL_RGB_BP_UNSIGNED_FLOAT: return getTranslatedFinalFormat();		//GL_COMPRESSED_RGB_BPTC_UNSIGNED_FLOAT
+				case gl::INTERNAL_RGB_BP_SIGNED_FLOAT: return getTranslatedFinalFormat();			//GL_COMPRESSED_RGB_BPTC_SIGNED_FLOAT
+				case gl::INTERNAL_RGB_BP_UNORM: return getTranslatedFinalFormat();					//GL_COMPRESSED_RGBA_BPTC_UNORM
+				case gl::INTERNAL_RGB_PVRTC_4BPPV1: return getTranslatedFinalFormat();				//GL_COMPRESSED_RGB_PVRTC_4BPPV1_IMG
+				case gl::INTERNAL_RGB_PVRTC_2BPPV1: return getTranslatedFinalFormat();				//GL_COMPRESSED_RGB_PVRTC_2BPPV1_IMG
+				case gl::INTERNAL_RGBA_PVRTC_4BPPV1: return getTranslatedFinalFormat();			//GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG
+				case gl::INTERNAL_RGBA_PVRTC_2BPPV1: return getTranslatedFinalFormat();			//GL_COMPRESSED_RGBA_PVRTC_2BPPV1_IMG
+				case gl::INTERNAL_RGBA_PVRTC_4BPPV2: return getTranslatedFinalFormat();			//GL_COMPRESSED_RGBA_PVRTC_4BPPV2_IMG
+				case gl::INTERNAL_RGBA_PVRTC_2BPPV2: return getTranslatedFinalFormat();			//GL_COMPRESSED_RGBA_PVRTC_2BPPV2_IMG
+				case gl::INTERNAL_ATC_RGB: return getTranslatedFinalFormat();						//GL_ATC_RGB_AMD
+				case gl::INTERNAL_ATC_RGBA_EXPLICIT_ALPHA: return getTranslatedFinalFormat();		//GL_ATC_RGBA_EXPLICIT_ALPHA_AMD
```

Comment:
why are you not solving for the correct format here (and below) @AnastaZIuk ?

Reply:
we don't have it in E_FORMAT as far as I know :D or maybe it is called different there

Reply:
all of these formats are actually these ones
https://github.com/buildaworldnet/IrrlichtBAW/blob/master/include/irr/asset/format/EFormat.h#L152

Reply:
to do

Reply:
well, I have to add them on writer yet